### ISSUE TITLE
editor-square-inputs: Dynamic disable doesn't immediately affect "Make a Block" modal

### ISSUE THREAD

**Issue by mybearworld** [ScratchAddons/ScratchAddons#7189]
### Description

`editor-square-inputs` affects inputs in the "Make a Block" modal. When dynamically enabling or disabling the addon, this modal isn't affected. You have to close out of it and open it back up to see the changes.

### Steps to reproduce

1. Open the "Make a block" modal
2. Dynamically enable or disable `editor-square-inputs`

### Environment

Browser: Edge 121
Scratch Addons version: 1.37.0
Operating system: Windows

**Comment by mybearworld** [comment]
This also happens with `editor-dark-mode` and `custom-block-shape`. Not with `custom-block-text`, though.

**Comment by WorldLanguages** [comment]
> This also happens with `editor-dark-mode` and `custom-block-shape`. Not with `custom-block-text`, though.

Nice to know.

**Comment by Joeclinton1** [comment]
Probably getting the wrong workplace when it calls:  updateAllBlocks(vm, addon.tab.traps.getWorkspace(), ScratchBlocks);

**Comment by WorldLanguages** [comment]
@Joeclinton1 Good point. I wonder if we should extend `updateAllBlocks` so that it receives both `addon.tab.traps.getBlockly().getMainWorkspace()` and `addon.tab.traps.getWorkspace()`.
If it receives the same workspace twice, it would continue to work as it does now.
If it receives two different workspaces (the actual, main one; and the custom block modal), it should run the updating routine on both of them.
Does this make sense?

**Comment by WorldLanguages** [comment]
To make matters more confusing, I believe it is always true that

```js
window.Blockly.getMainWorkspace() === addon.tab.traps.getBlockly().getMainWorkspace()
```

but I'm not 100% sure.

**Comment by mxmou** [comment]
> @Joeclinton1 Good point. I wonder if we should extend `updateAllBlocks` so that it receives both `addon.tab.traps.getBlockly().getMainWorkspace()` and `addon.tab.traps.getWorkspace()`.

Blockly stores all workspaces in an object called `Blockly.Workspace.WorkspaceDB_`. Always updating every workspace would probably be more reliable.

**Comment by WorldLanguages** [comment]
I wonder if there's a maximum amount of workspaces that can be stored in `WorkspaceDB_` or if it just keeps growing as you use the editor...?

**Comment by mxmou** [comment]
> I wonder if there's a maximum amount of workspaces that can be stored in `WorkspaceDB_` or if it just keeps growing as you use the editor...?

The custom block editing workspace is removed when the modal is closed:
![image](https://github.com/ScratchAddons/ScratchAddons/assets/51849865/2a1880fc-a27a-48b1-b197-1185cf35e44b)
